simpeg_drivers/driver.py

```diff
@@ -323,6 +323,15 @@ def out_group(self) -> SimPEGGroup:
         """
         return self._out_group
 
+    @out_group.setter
+    def out_group(self, value: SimPEGGroup):
+        if not isinstance(value, SimPEGGroup):
+            raise TypeError("Output group must be a SimPEGGroup.")
+
+        self.params.out_group = value
+        self.params.update_out_group_options()
+        self._out_group = value
```

Comment:
why creating a double reference here?
self.params.out_group and self._out_group are the same ?!